src/test/java/org/broadinstitute/hellbender/utils/codecs/gtf/EnsemblGtfCodecUnitTest.java

```diff
@@ -16,8 +19,96 @@
  */
 public class EnsemblGtfCodecUnitTest extends GATKBaseTest {
 
-    private static final String testResourceDir = publicTestDir + "org/broadinstitute/hellbender/utils/codecs/gtf/";
-    private static final String eColiTestDir = publicTestDir + "org/broadinstitute/hellbender/tools/funcotator/ecoli_ds/gencode/ASM584v2/";
+    private static final String testResourceDir           = publicTestDir + "org/broadinstitute/hellbender/utils/codecs/gtf/";
+    private static final String eColiTestDir              = publicTestDir + "org/broadinstitute/hellbender/tools/funcotator/ecoli_ds/gencode/ASM584v2/";
+    private static final String ECOLI_UCSC_GENOME_VERSION = "ASM584v2";
+    private static final String ECOLI_CONTIG_NAME         = "Chromosome";
+
+    private GencodeGtfFeature createEcoliEnsemblGene(final int startingFeatureOrder, final String geneId, final String geneName, final int geneStart,
+                                             final int geneEnd, final String transcriptId, final String transcriptName,
+                                             final String exonId, final int cdsStart, final int cdsEnd) {
+
+        // Placeholder for constant extra data:
+        final String geneAnonymousOptionalFields = " gene_source \"ena\";";
+        final String transcriptAnonymousOptionalFields = geneAnonymousOptionalFields + " transcript_source \"ena\";";
+
+        int featureOrderNum = startingFeatureOrder;
+
+        GencodeGtfFeatureBaseData data;
+
+        data = new GencodeGtfFeatureBaseData(EnsemblGtfCodec.GTF_FILE_TYPE_STRING, featureOrderNum++, ECOLI_CONTIG_NAME, GencodeGtfFeature.AnnotationSource.ena, GencodeGtfFeature.FeatureType.GENE,
```

Comment:
ugh, yeah, I can see why this was a pain

Reply:
It wasn't so bad when I figured out the format of the ecoli files.  the helper methods really made it much easier.